src/plugins/workspace/public/components/workspace_menu/workspace_menu.tsx

```diff
@@ -3,62 +3,73 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import './workspace_menu.scss';
 import { i18n } from '@osd/i18n';
-import React, { useState } from 'react';
+import React, { useMemo, useState } from 'react';
 import { useObservable } from 'react-use';
 import {
+  EuiAvatar,
+  EuiButton,
+  EuiButtonEmpty,
   EuiButtonIcon,
   EuiContextMenu,
   EuiFlexGroup,
   EuiFlexItem,
-  EuiIcon,
-  EuiListGroup,
-  EuiListGroupItem,
+  EuiPanel,
   EuiPopover,
   EuiText,
 } from '@elastic/eui';
-import type { EuiContextMenuPanelItemDescriptor } from '@elastic/eui';
-
+import { FormattedMessage } from '@osd/i18n/react';
+import { BehaviorSubject } from 'rxjs';
 import {
   WORKSPACE_CREATE_APP_ID,
   WORKSPACE_LIST_APP_ID,
+  MAX_WORKSPACE_PICKER_NUM,
   WORKSPACE_DETAIL_APP_ID,
 } from '../../../common/constants';
-import { cleanWorkspaceId, formatUrlWithWorkspaceId } from '../../../../../core/public/utils';
+import { formatUrlWithWorkspaceId } from '../../../../../core/public/utils';
 import { CoreStart, WorkspaceObject } from '../../../../../core/public';
+import { getUseCaseFromFeatureConfig } from '../../utils';
+import { recentWorkspaceManager } from '../../recent_workspace_manager';
+import { WorkspaceUseCase } from '../../types';
 
+const defaultHeaderName = i18n.translate(
+  'core.ui.primaryNav.workspacePickerMenu.defaultHeaderName',
```

Comment:
the namespace/prefix should be workspace not core as the code are workspace folder

Reply:
updated